### DISCUSSION TITLE
Other CMSIS Standards

### DISCUSSION THREAD

**Discussion by 0Grit** [espressif/svd#17] [General]
Is there any interest in taking up other standards from CMSIS?
ARM is actually opening up CMSIS in certain areas through Linaro.
The Open-CMSIS-packs project for example has public weekly meetings and I've been told is open to anybody.

https://github.com/Open-CMSIS-Pack/Open-CMSIS-Pack/discussions
https://linaro.atlassian.net/wiki/spaces/CMSIS/overview
https://www.open-cmsis-pack.org/

**Comment by jessebraham** [comment]
_(Original conversation can be found in this issue: https://github.com/espressif/svd/issues/15)_

Please see my latest post in the aforementioned issue. In short, we do not have any plans to take up any further CMSIS standards at this point. This may or may not change in the future.

**Comment by ilg-ul** [comment]
CMSIS stands for **Common Microcontroller Software Interface Standard**, but by *microcontroller* Arm understands Cortex devices.

RISC-V does not need to adopt CMSIS as-is, but if they will not try to standardise somehow the software side, the whole world of RISC-V microcontrollers will be a mess.

The hardware guys already made a mess with the too many possible combinations of extensions, which must be supported by bare-metal toolchains. 

In the latest release of the xPack RISC-V Embedded GCC, I had to implement almost 100 different multi-libs, and the distribution grew to about 500 MB, which is insane.

The same will happen with the specifications for the new interrupt controller, which define lots of possible different operating modes, such that portable system libraries will become extremely complicated to support them all.

**Comment by ilg-ul** [comment]
RISC-V part of Open CMSIS Pack? It's not going to happen. It is an ST thing, anyway.

**Reply by 0Grit** [reply]
I realize the 'conflicts of interest' but I tend to learn things and meet interesting people when I poke boundaries.
